packages/openssl-1.1/build.sh

```diff
@@ -0,0 +1,113 @@
+		CFLAGS+=" -fno-integrated-as"
+	fi
+
+	perl -p -i -e "s@TERMUX_CFLAGS@$CFLAGS@g" Configure
+	test $TERMUX_ARCH = "arm" && TERMUX_OPENSSL_PLATFORM="android-arm"
+	test $TERMUX_ARCH = "aarch64" && TERMUX_OPENSSL_PLATFORM="android-arm64"
+	test $TERMUX_ARCH = "i686" && TERMUX_OPENSSL_PLATFORM="android-x86"
+	test $TERMUX_ARCH = "x86_64" && TERMUX_OPENSSL_PLATFORM="android-x86_64"
+	./Configure $TERMUX_OPENSSL_PLATFORM \
+		--prefix=$TERMUX_PREFIX \
+		--openssldir=$TERMUX_PREFIX/etc/tls \
+		shared \
+		zlib-dynamic \
+		no-ssl \
+		no-hw \
+		no-srp \
+		no-tests
+}
+
+termux_step_make() {
+	make depend
+	make -j $TERMUX_MAKE_PROCESSES all
+}
+
+termux_step_make_install() {
+	# "install_sw" instead of "install" to not install man pages:
+	make -j 1 install_sw MANDIR=$TERMUX_PREFIX/share/man MANSUFFIX=.ssl
+
+	mkdir -p $TERMUX_PREFIX/etc/tls/
+
+	cp apps/openssl.cnf $TERMUX_PREFIX/etc/tls/openssl.cnf
+
+	install -m755 -d $TERMUX_PREFIX/include/openssl-1.1
+	mv $TERMUX_PREFIX/include/openssl $TERMUX_PREFIX/include/openssl-1.1/
+	mv $TERMUX_PREFIX/bin/openssl $TERMUX_PREFIX/bin/openssl-1.1
+
+	install -m755 -d $TERMUX_PREFIX/lib/openssl-1.1
+	rm $TERMUX_PREFIX/lib/libcrypto.so $TERMUX_PREFIX/lib/libssl.so
+	mv $TERMUX_PREFIX/lib/libcrypto.so.1.1 $TERMUX_PREFIX/lib/openssl-1.1
+	mv $TERMUX_PREFIX/lib/libssl.so.1.1 $TERMUX_PREFIX/lib/openssl-1.1
+	mv $TERMUX_PREFIX/lib/libcrypto.a $TERMUX_PREFIX/lib/openssl-1.1
+	mv $TERMUX_PREFIX/lib/libssl.a $TERMUX_PREFIX/lib/openssl-1.1
+	
+	install -m755 -d $TERMUX_PREFIX/lib/pkgconfig/openssl-1.1
+	mv $TERMUX_PREFIX/lib/pkgconfig/openssl.pc $TERMUX_PREFIX/lib/pkgconfig/openssl-1.1
+	mv $TERMUX_PREFIX/lib/pkgconfig/libssl.pc $TERMUX_PREFIX/lib/pkgconfig/openssl-1.1
+	mv $TERMUX_PREFIX/lib/pkgconfig/libcrypto.pc $TERMUX_PREFIX/lib/pkgconfig/openssl-1.1
+
+
+	sed -e 's|/include$|/include/openssl-1.1|' -i $TERMUX_PREFIX/lib/pkgconfig/openssl-1.1/*.pc
+}
+
+termux_step_post_make_install() {
+	test -d $TERMUX_PREFIX/include/openssl.tmp && mv $TERMUX_PREFIX/include/openssl{.tmp,} || :
+
+	test -a $TERMUX_PREFIX/lib/pkgconfig/libcrypto.pc.tmp && mv $TERMUX_PREFIX/lib/pkgconfig/libcrypto.pc{.tmp,} || :
+	test -a $TERMUX_PREFIX/lib/pkgconfig/libssl.pc.tmp && mv $TERMUX_PREFIX/lib/pkgconfig/libssl.pc{.tmp,} || :
+	test -a $TERMUX_PREFIX/lib/pkgconfig/openssl.pc.tmp && mv $TERMUX_PREFIX/lib/pkgconfig/openssl.pc{.tmp,} || :
+
+	test -a $TERMUX_PREFIX/lib/libcrypto.so.tmp && mv $TERMUX_PREFIX/lib/libcrypto.so{.tmp,} || :
+	test -a $TERMUX_PREFIX/lib/libssl.so.tmp && mv $TERMUX_PREFIX/lib/libssl.so{.tmp,} || :
+	test -a $TERMUX_PREFIX/lib/libcrypto.a.tmp && mv $TERMUX_PREFIX/lib/libcrypto.a{.tmp,} || :
+	test -a $TERMUX_PREFIX/lib/libssl.a.tmp && mv $TERMUX_PREFIX/lib/libssl.a{.tmp,} || :
+
+}
+
+termux_step_post_massage() {
+	rm -rf include/openssl
+	rm -f lib/pkgconfig/libcrypto.pc
+	rm -f lib/pkgconfig/libssl.pc
+	rm -f lib/pkgconfig/openssl.pc
+	rm -f lib/libcrypto.so
+	rm -f lib/libssl.so
+	rm -f lib/libcrypto.a
```

Comment:
@Grimler91, this line has no effect because termux_step_post_massage is run after subpackages are built in `termux_step_massage`

Reply:
Add the files to TERMUX_PKG_RM_AFTER_INSTALL instead, that should work

Reply:
Or maybe not if plan is restore them and then remove them from package

Reply:
> Add the files to TERMUX_PKG_RM_AFTER_INSTALL instead, that should work

That would work in the short run, I'll go by this for now. I have some other ways of fixing it too in my brain, but I would like to do this after this is merged

Reply:
openssl seem to have a more or less undocumented `--libdir` configure arg, so you can configure `--libdir=$TERMUX_PREFIX/lib/openssl-1.1` instead of backing up files (`test -a $TERMUX_PREFIX/lib/pkgconfig/libcrypto.pc && mv $TERMUX_PREFIX/lib/pkgconfig/libcrypto.pc{,.tmp} || :`) and then moving built files to correct place (`mv $TERMUX_PREFIX/lib/libcrypto.so.1.1 $TERMUX_PREFIX/lib/openssl-1.1`).

Doesn't seem to be similar options for bindir and includedir though

Reply:
Thanke